tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs

```diff
@@ -67,7 +67,8 @@ private static Obj GetConfigObj(string rulesFile, List<Obj> argCache, Encoder en
                 var root = JToken.ReadFrom(jsonReader);
 
                 LogRuleDetailsIfDebugEnabled(root);
-                configObj = encoder.Encode(root, argCache);
+                // applying safety limits during rule parsing could result in trucated rules
+                configObj = encoder.Encode(root, argCache, applySafetyLimits: false);
```

Comment:
If we never set this to `true`, we may as well delete those code paths right?

Reply:
it's set to true here: https://github.com/DataDog/dd-trace-dotnet/pull/2516/files#diff-805142da125346ffbe68e5f28cbf29ab92ff95922994806d248020be9b6250c8R37

There's two uses of this function:
- parse rule file - don't truncate as this could result in misbehaviour of rules
- incoming data from user - truncate as this could result in a payload that's too large for the waf